aws/resource_aws_elasticache_cluster.go

```diff
@@ -210,6 +217,81 @@ func resourceAwsElasticacheCluster() *schema.Resource {
 		},
 
 		Schema: resourceSchema,
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				// Plan time validation for az_mode
+				// InvalidParameterCombination: Must specify at least two cache nodes in order to specify AZ Mode of 'cross-az'.
+				azMode, azModeOk := diff.GetOk("az_mode")
```

Comment:
Not something to hold up merging, and probably leaning more towards preference, but you could rewrite this a little so you could just reuse the standard `ok` and short circuiting a little earlier (also helping the line length a little):

```golang
if azMode, ok := diff.GetOk("az_mode"); !ok || azMode.(string) != elasticache.AZModeCrossAz {
  return nil
}

if numCacheNodes, ok := diff.GetOk("num_cache_nodes"); !ok || numCacheNodes.(int) != 1 {
  return nil
}

return errors.New(...)
```